components/brave_wallet_ui/components/desktop/views/portfolio/components/token-lists/token-list.tsx

```diff
@@ -410,15 +416,43 @@ export const TokenLists = ({
     setSearchValue('')
   }, [])
 
-  const listUiByNetworks = React.useMemo(() => {
-    if (!networks) {
+  const noNetworks = !networks || networks.length === 0
+  const noAccounts = !accounts || accounts.length === 0
+  const showEmptyState = noNetworks || noAccounts
+
+  const sortedNetworksWithBalances = React.useMemo(() => {
+    if (noNetworks) {
       return undefined
     }
-    return [...networks].sort((a, b) => {
-      const aBalance = getNetworkFiatValue(a)
-      const bBalance = getNetworkFiatValue(b)
-      return bBalance.minus(aBalance).toNumber()
-    })?.map((network) => {
+    return [...networks].filter(
+      (network) => getNetworkFiatValue(network)
+        .gt(0)).sort((a, b) => {
+          const aBalance = getNetworkFiatValue(a)
+          const bBalance = getNetworkFiatValue(b)
+          return bBalance.minus(aBalance).toNumber()
+        })
```

Comment:
nit: do you think this looks any easier to read?
```suggestion
    return [...networks].filter((network) => {
      return getNetworkFiatValue(network).gt(0)
    }).sort((a, b) => {
      const aBalance = getNetworkFiatValue(a)
      const bBalance = getNetworkFiatValue(b)
      return bBalance.minus(aBalance).toNumber()
    })
```